crates/compiler/roc_target/src/lib.rs

```diff
@@ -9,7 +9,8 @@ use target_lexicon::Triple;
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum OperatingSystem {
     Windows,
-    Unix,
+    Linux,
+    MacOS,
     Wasi,
 }
```

Comment:
I'm not completely sold on this solution so I am open to doing something else.

Reply:
Alternatively I could pass a boolean into `link_macos` and `link_linux` that determines whether or not the extension should be overridden. But that seems like a bandaid.